sqlmesh/core/snapshot/evaluator.py

```diff
@@ -142,6 +141,11 @@ def apply(query_or_df: QueryOrDF, index: int = 0) -> None:
             **common_render_kwargs,
         )
 
+        if not snapshot.is_model:
```

Comment:
Why wait so long, why not return at the top?

Reply:
I can't recall why. Will move to the top.